packages/derived-wallet-base/src/abstraction.ts

```diff
@@ -1,43 +1,15 @@
-import { AccountAddress, AuthenticationKey, hashValues, isValidFunctionInfo, Serializer } from '@aptos-labs/ts-sdk';
+import { AuthenticationKey, DerivableAbstractedAccount } from '@aptos-labs/ts-sdk';
 
 /**
- * The domain separator used to calculate the DAA account address.
- */
-export const ADDRESS_DOMAIN_SEPARATOR = 5;
-
-/**
- * Compute the account address of the DAA
- * The DAA account address is computed by hashing the function info and the account identity
- * and appending the domain separator (5)
- *
  * @param functionInfo - The authentication function
  * @param accountIdentifier - The account identity
  * @returns The account address
  */
-export function computeDomainAuthenticationKey(functionInfo: string, accountIdentifier: Uint8Array) {
-  if (!isValidFunctionInfo(functionInfo)) {
-    throw new Error(`Invalid authentication function ${functionInfo}`);
+export function computeDomainAuthenticationKey(functionInfo: string, accountIdentifier: Uint8Array): AuthenticationKey {
+  try{
+    const add = DerivableAbstractedAccount.computeAccountAddress(functionInfo, accountIdentifier);
+    return new AuthenticationKey({ data:add });
+  } catch (e) {
+    throw new Error(`Error computing domain authentication key: ${e}`);
```

Comment:
Why are we catching and throwing a different error?
We lose information about the stack trace etc..

How about letting the consumer catch the underlying error instead?